style.css

```diff
@@ -1,19 +1,84 @@
+:root {
+	--primary-background-color: hsl(24 20% 95%);
+	--primary-foreground-color: hsl(24 20% 5%);
+	--secondary-background-color: hsl(200 10% 20%);
+	--secondary-foreground-color: white;
+	--tertiary-background-color: transparent;
+	--tertiary-foreground-color: hsl(200 10% 20%);
+	--primary-shadow-color: white;
+	--secondary-shadow-color: hsl(0 0% 0% / .5);
+	--primary-border-color: hsl(0 0% 0% / .25);
+	--secondary-border-color: var(--secondary-background-color);
+	--shade-color: hsl(0 0% 0% / .3);
+
+	--spec-link-background-color: var(--secondary-background-color);
+	--spec-link-foreground-color: var(--secondary-foreground-color);
+	--highlight-background-color: #f06;
+	--highlight-foreground-color: white;
+	--prefix-background-color: var(--shade-color);
+	--prefix-foreground-color: var(--spec-link-foreground-color);
+	--progress-color: var(--shade-color);
+
+	--pass-color: #490;
+	--almost-pass-color: #8c0;
+	--slightly-buggy-color: #bb0;
+	--buggy-color: #e80;
+	--very-buggy-color: #e40;
+	--fail-color: #e20;
+	--epic-fail-color: #b00;
+
+	--aside-border-color: hsl(0 0% 80% / .8);
+	--aside-background-color: hsl(0 0% 98% / .6);
+	--aside-foreground-color: black;
+}
+
+@media (prefers-color-scheme: dark) {
```

Comment:
How will we do the user switch without duplicating this entire block?

Reply:
Well, the block depends on either the media query above or some kind of manual switch. And as far as I know, there is currently no pure CSS way that covers both.

Style queries as well as `@when` / `@else` may come in handy here once they exist. Then we only need to set a custom property and the rest would work in CSS.

We might achieve that by doing the checks in JS and then setting the custom properties by dynamically injecting a stylesheet but that feels a bit hacky.
Though if you have an idea how to do it better, please let me know!

Anyway, I'd say let's stick to this solution for now and discuss the user switch separately.

Reply:
If we have a `--dark` variable that goes from 0 to 1, we can easily have conditionals on it for anything numerical, like so:

```css
--not-dark: calc( 1 - var(--dark)); 
property: calc(var(--dark) * value_if_dark + var(--not-dark) * value_if_not_dark);
```

Reply:
Yes, a variable like that could be used to switch between the colors. Though that approach still feels hacky and requires the media query to be checked in JS.